server/controllers/stats.js

```diff
@@ -1,13 +1,127 @@
 var express = require('express');
 var router = express.Router();
 var helpers = require('../helpers/app-helpers.js');
+var logger = require('config-logger');
 
 router.get('/', function(req, res) {
   successcb = function(data){
-    res.json(data);
+    // convert the raw Prometheus data to JSON in a form usable by the client
+
+    var queuedLabel = 'fn_queued';
+    var runningLabel = 'fn_running';
+    var completedlabel = "fn_completed";
+    var failedLabel = "fn_failed";
+    var appNameMetric = "fn_appname";
+    var pathNameMetric = "fn_path";
+
+    // Construct a RE to extract data from the raw Prometheus metrics data, which will include lines like
+    // fn_completed{fn_appname="hello-cold-async-a",fn_path="/hello-cold-async-a2"} 21
+
+    // metricNameRE matches any of the required metric names, and saves the metric name
+    var metricNameRE = '('+queuedLabel+'|'+runningLabel+'|'+completedlabel+'|'+failedLabel+')'; 
+
+    // labelNameRE matches a label name, which is anything that's not a = (equals sign), and saves the label name
+    var labelNameRE = '([^=]+)';   
+    // labelValueRE matches a label value, which is anything that's not a " (double quote), and save the label value
+    var labelValueRE = '\"([^\"]*)\"'; 
+    // labelRE is a combined label name/value pair
+    var labelRE = labelNameRE+'='+labelValueRE;
+
+    // spacesRE matches one or more spaces
+    var spacesRE = '\\s+';            
+    //  valueRE matches one or more numeric digits, and saves the metric value
+    var valueRE = '(\\d+)'             
+    
+    // allMetricsRE is the complete RE (allowing for more than two labels)
+    var allMetricsRE    = metricNameRE+   '{'+labelRE+'(?:,'+labelRE+')+}'+spacesRE+valueRE;
```

Comment:
`metricNameRE` is at the start of a line, so I will prepend a `^` to the start of the RE to handle this (This means I need to add the `m` flag when the RE is executed, so that this multiline input string is treated as multiple lines).

Reply:
yes, sorry I've actually meant '^' not '{'. :-)